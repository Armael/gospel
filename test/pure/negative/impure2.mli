val f : int -> int
(*@ y = f x
    pure
    diverges *)

(* EXPECTED
   [125] File "impure2.mli", line 2, characters 5-6:
         Error: Type checking error: a pure function cannot diverge.
*)
