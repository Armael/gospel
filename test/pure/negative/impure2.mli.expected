
*******************************
********** Parsed file ********
*******************************
val f : int -> int[@@gospel {| y = f x
    pure
    diverges |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

val f : int -> int
(*@ y = f x
    diverges
     *)
File "impure2.mli", line 5, characters 21-22:
Error: Type checking error: a pure function cannot diverge
